modules/scripts/genmainindex.py

```diff
@@ -0,0 +1,19 @@
+import os
+
+toctree = []
+for (dirpath, dirnames, filenames) in os.walk(os.curdir):
+    caption_template = '   :caption: {caption}'
+    content_template = '   {document}'
+    if 'index.rst' in filenames:
+        caption = dirpath[2:]
```

Comment:
repository are clonned as project name. i'm generating toc tree using the `genindex.py` for each repository and then i'm traversing each folder for any index.rst files is there are not. if index.rst is there means then it is a repository and I'm generating toc tree by taking that index.rst's folder name as structure.